### DISCUSSION TITLE
How to Upload Files to a Specific Directory in a Serverless Container using Serverpod?

### DISCUSSION THREAD

**Discussion by iampopal** [serverpod/serverpod#2859] [Q&A] [answered]
Hello Serverpod Developers,

We are in the process of setting up a filter within a container that will run in a serverless environment. The storage will be mounted, ensuring that even if the server is restarted, no data will be lost.

I have already reviewed the documentation regarding file uploads (link: https://docs.serverpod.dev/concepts/file-uploads), but I couldn't find a clear method for uploading files into a specific directory within the server.

Could you guide me on whether there is a possible or simple way to achieve this? Any insights or examples would be greatly appreciated.

Thanks in advance for your help!

**Comment by SandPod** [comment, marked as answer]
Currently, there is no module to upload directly to the server directory, Serverpod supports file uploads to the database or to an AWS or GCP bucket.

There is, however a PR that implements support for this: https://github.com/serverpod/serverpod/pull/2571

**Comment by iampopal** [comment]
The GCP issue is that we cannot verify is the file uploaded or not. What
will be best practice that to verify the file exists there.

And

Calling the gcp link from user app exposes the gcs link to public and it
might become vulnerability in future.

I want to give permission to user so each user only view their own files
how can i achieve this with gcs.

Thank you.

**Comment by iampopal** [comment]
@SandPod any idea?

**Reply by SandPod** [reply]
This is also a work in progress with the PR available here: https://github.com/serverpod/serverpod/pull/2745

It sounds like you would like to use signed URLs to provide time-limited access to a resource: https://cloud.google.com/storage/docs/access-control/signed-urls

**Reply by iampopal** [reply]
Thank you, 

If possible to provide the signed url way an example so we can go securely without public folder in cloud storage

**Reply by iampopal** [reply]
....?
